plugins/modules/ecs_ecr.py

```diff
@@ -550,7 +581,8 @@ def main():
         purge_policy=dict(required=False, type='bool'),
         lifecycle_policy=dict(required=False, type='json'),
         purge_lifecycle_policy=dict(required=False, type='bool'),
-        scan_on_push=(dict(required=False, type='bool', default=False))
+        scan_on_push=(dict(required=False, type='bool', default=False)),
+        encryption_configuration=(dict(required=False, type='dict'))
```

Comment:
Please also define what's supposed to be in this dict, It's relatively easy, just adding in the 'options' dict.  Retroactively applying this can be messy, since you go from the state of garbage being dropped on the floor, to garbage causing errors.

Something like:
```
  encryption_configuration=(
    dict(
      required=False,
      type='dict',
      options=dict(
        encryption_type=dict(required=True, type='str', choices=['AES256', 'KMS'],)
        kms_key=dict(required=False, type='str')
      ),
      required_if=('encryption_type', 'KMS', ['kms_key']),
    ),
  ),
```

Note: you can also add things like dependencies/conflicts between the sub-options.